libnetwork/drivers/bridge/setup_ip_tables_linux_test.go

```diff
@@ -374,3 +377,68 @@ func TestOutgoingNATRules(t *testing.T) {
 		})
 	}
 }
+
+func TestMirroredWSL2Workaround(t *testing.T) {
+	// Avoid overwriting a real "/usr/bin/wslinfo" (and clashing with a real
+	// loopback0).
+	if _, err := os.Stat(wslinfoPath); err == nil {
+		t.Skip("Skipping test because " + wslinfoPath + " exists")
+	}
```

Comment:
Sounds like a job for namespaces! We don't have to worry about clashing with a real loopback0 because the test cases are invoked in an unshared network namespace thanks to `netnsutils.SetupTestOSContext()`. While we don't have a ready-made test utility for unsharing a mount namespace and setting things up such that parts of the host filesystem can be mutated such that the mutations are only visible to the test, we do have all the building blocks. The general idea is to over-mount a path with an overlayfs with the same path as the lowerdir, and a tmpfs path as the upper:
```console
root@56e9443a7364:~# mkdir /tmp/upper /tmp/work
root@56e9443a7364:~# unshare --mount-proc --propagation slave bash
root@56e9443a7364:~# mount -t overlay -olowerdir=/usr/bin,upperdir=/tmp/upper,workdir=/tmp/work overlay /usr/bin
root@56e9443a7364:~# rm /usr/bin/ls
root@56e9443a7364:~# ls -ld
bash: ls: command not found
root@56e9443a7364:~# exit
exit
root@56e9443a7364:~# ls -ld
drwx------ 1 root root 4096 Jun 28 23:05 .
```

...it may be too much to tack onto this PR, though. Save for a follow-up?

Reply:
I forgot to delete that check when I changed `wslinfoPath` from `const` to `var` so that the test could modify it to avoid the clash that way instead. It's gone now.